charts/judge/docs/kratos.md

```diff
@@ -0,0 +1,114 @@
+
+## Ingress Configuration
+
+Kratos exposes two ingress endpoints, one for administrative purposes and the other for public access:
+
+### Public Ingress
+
+- **Host:** kratos.testifysec.localhost
+- **TLS Configuration:** Uses the `kratos-tls-secret` for HTTPS encryption
+
+### Admin Ingress
+
+- **Host:** kratos.admin.local.com
+- **TLS Configuration:** Not enabled by default
+
+## Application-Specific Configuration
+
+Kratos offers various configuration options tailored to different use cases and environments. Below are the key configuration parameters:
+
+### Database Configuration
+
+- **DSN:** Specifies the database connection string for Kratos.
+- **Automigration:** Controls database schema initialization and migration.
+
+### Identity Schemas
+
+You can define multiple identity schemas to customize user attributes and validation rules.
+
+### Email Templates
+
+Customize the content of emails sent by Kratos for actions like account recovery and verification.
+
+### Serve Configuration
+
+Configures the ports and base URL for serving Kratos endpoints, along with CORS settings.
+
+### Self-Service Flows
+
+Defines URLs and behavior for user self-service actions like account recovery and verification.
+
+### OIDC Methods Configuration
+
+Enables OIDC authentication methods and configures providers like GitHub and GitLab.
+
+## Integrating with GitLab and GitHub
+
+To integrate Kratos with GitLab and GitHub, you'll need to generate application IDs and keys for JUDGE. Here's how:
+
+1. **GitHub Integration**:
+   - Log in to your GitHub account.
+   - Go to Settings > Developer settings > OAuth Apps.
+   - Click on "New OAuth App" and fill in the required details.
+   - Set the Authorization callback URL to your Kratos svc, for example `https://kratos.testifysec.localhost/selfservice/methods/oidc/callback/github`.
+   - After creating the app, note down the Client ID and Client Secret.
+   - Update the Kratos configuration with these values for the GitHub provider.
+
+2. **GitLab Integration**:
+   - Log in to your GitLab account.
+   - Go to Settings > Applications.
+   - Create a new application and fill in the required details.
+   - Set the Redirect URI to your Kratos svc, for example `https://kratos.testifysec.localhost/selfservice/methods/oidc/callback/gitlab`.
+   - Once created, obtain the Application ID and Secret.
+   - Update the Kratos configuration with these values for the GitLab provider.
+
+After generating your secret(s), in your values.yaml define:
+
+```yaml
+kratos:
+  kratos:
+    config:
+      selfservice:
+        allowed_return_urls:
+          - https://login.testifysec.localhost
+          - https://kratos.testifysec.localhost
+          - https://judge.testifysec.localhost
+          - https://localhost
+          - https://judge.testifysec.localhost:8077
```

Comment:
We should make sure this is templated to a configmap for the kratos deployment. I'm not sure what these values should be but, my guess is they will be related to the deployed DNS names of the service.